runtime/sam/expr/function/parse.go

```diff
@@ -1,79 +1,119 @@
 package function
 
 import (
+	"fmt"
 	"net/url"
 	"strconv"
 	"strings"
 
 	"github.com/brimdata/super"
+	"github.com/brimdata/super/zcode"
 	"github.com/brimdata/super/zio/zsonio"
 	"github.com/brimdata/super/zson"
 )
 
 // https://github.com/brimdata/super/blob/main/docs/language/functions.md#parse_uri
 type ParseURI struct {
-	zctx      *super.Context
-	marshaler *zson.MarshalZNGContext
+	zctx    *super.Context
+	typ     super.Type
+	builder zcode.Builder
```

Comment:
Nit: Rewriting this in terms of zcode.Builder doesn't seem necessary for adding parse_uri() to vam and makes makes the code more complicated.

Reply:
I had one this to support null return values but am now realizing I can still do this using the old method. I'll change it back.